design/accepted/kv_auth.md

```diff
@@ -0,0 +1,115 @@
+# Implementing Auth Package with KV Database
+
+This document describes the entities of lakeFS auth package, the relationship between them and offers
+a possible representation of them in the KV database.
+
+## Entities
+
+### User
+
+- Can be looked up by `ID`, `Username` & `Email`.
+- Deleted by `Username`.
+- Listed by `Username`.
+
+### Group
+
+- Looked up, deleted and listed by `DisplayName`.
+
+### Policies
+
+- Looked up, deleted and listed by `DisplayName`.
+
+### Credentials
+
+- Has a foreign id reference to `User.ID`.
+    - The `User.ID` is used for looking up the `User.Username`.
+    - Almost all actions are in the context of `User.Username`, i.e. it is always passed.
+- The only action without a `User.Username` passed is credentials lookup by `AccessKeyID`.
+
+
+## Entities Relationship
+
+### Group <-> User
+
+- Add/Remove `User` to `Group` by `User.Username` & `Group.DisplayName`. 
+- List all the `Group`s for a `User` by `User.Username`.
+- List all the `Users`s in a `Group` by `Group.DisplayName`.
+
+### Policy <-> User
+
+- Attach/Detach `Policy` to `User` by `User.Username` & `Policy.DisplayName`.
+- List all `Policies` for a `User` by `User.Username`.
+- List all effective `Policies` for a `User` by `User.Username`, this includes all `Policies` for `Group`s is a member of.
+
+### Policy <-> Group
+
+- Attach/Detach `Policy` to `Group` by `Group.DisplayName` & `Policy.DisplayName`.
+- List all `Policies` for a `Group` by `Group.DisplayName`.
+    - `Group` cannot be a member of another `Group`, therefore all effective policies are attached to it directly.
+
+## Representation in the KV world
```

Comment:
In addition to the keys, I think it will be very very helpful to explicitly describe the KV model structs

Reply:
We have a separate issue for that #3349